### PR TITLE
feat: Elastic training

### PR DESCRIPTION
**Pull request by shjwudp**
Major change:

1. Standalone elastic launch script, only support PyTorch >= 1.9.0
2. The startup timing of AutotuneServer is changed from distributed.launch to init_process_group
3. Some installation dependencies are added to setup.py

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def check_torch_version():
             "prometheus_client",
             "parallel-ssh",
             "pydantic",
+            "setuptools_rust",
+            "tqdm",
+            "colorama",
+            "requests",
```
Comment (NOBLES5E) [inline setup.py:50]: don't add these deps to bagua, rust related deps are already in `bagua-core`

File: bagua/distributed/run.py [hunk 2]
```diff
@@ -0,0 +1,598 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (NOBLES5E) [inline bagua/distributed/run.py:3]: also add our license notice

File: setup.py (hunk 1 above)
Comment (shjwudp) [reply setup.py]: OK, Useless dependencies have been removed. `requests` need to stay.

File: bagua/distributed/run.py (hunk 2 above)
Comment (shjwudp) [reply bagua/distributed/run.py:5]: DONE

**Pull request merged**